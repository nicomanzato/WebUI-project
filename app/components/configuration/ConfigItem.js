'use strict';

import React from 'react';
import {StyleSheet, Text, View, CheckBox} from 'react-native';

class ConfigItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: this.props.value,
    }

    this.handleOnValueChange = this.handleOnValueChange.bind(this);
  }

  handleOnValueChange() {
    this.setState({
      value: !this.state.value,
    });

    this.props.onChange();
  }

  render() {
    return (
     <View style={styles.configItem}>
       <View style={styles.configTextView}>
         <Text style={styles.configText}>{this.props.title}</Text>
       </View>
       <View style={styles.checkboxView}>
         <CheckBox
          style={styles.configCheckBox}
          onValueChange={this.handleOnValueChange}
          value={this.state.value}
         />
       </View>
     </View>
    );
  }
}

export default ConfigItem;

const styles = StyleSheet.create({
  configItem: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderColor: "#ccc",
    padding: 10,
  },
  configText: {
    fontSize: 17,
  },
  configCheckBox: {},
  configTextView: {
    flex: 0.9,
    height: 50,
  },
  checkboxView: {
    flex: 0.1,
    height: 50,
  }
});
