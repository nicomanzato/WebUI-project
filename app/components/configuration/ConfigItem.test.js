import React from 'react';
import ConfigItem from './configItem'
import {CheckBox} from 'react-native'

import { shallow, mount, ReactWrapper } from 'enzyme';

describe('config item', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <ConfigItem
        title='Test title'
        onChange={() => {}}
        value={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange when pressed', () => {

    const onChange = jest.fn().mockName('mockedFunction');

    const wrapper = mount(
      <ConfigItem
        title='Test title'
        onChange={onChange}
        value={false}
      />
    );

    const configItem = wrapper.find(CheckBox).first();

    configItem.props().onValueChange();

    expect(onChange).toHaveBeenCalled();
  });

});
