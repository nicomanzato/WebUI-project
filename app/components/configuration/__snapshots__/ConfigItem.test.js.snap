// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config item should render correctly 1`] = `
<Component
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#ccc",
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 0.9,
        "height": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Test title
    </Text>
  </Component>
  <Component
    style={
      Object {
        "flex": 0.1,
        "height": 50,
      }
    }
  >
    <UnimplementedView
      onValueChange={[Function]}
      style={Object {}}
      value={false}
    />
  </Component>
</Component>
`;
